File: generators/fullstack-app/templates/2.1.0:7.1.0/<%= projectName %>-frontend/src/app/@lib/umari/datafilter/core/udf-aggregable.ts
/**
 * Informação das agregações à serem feitas, oriundo do LoadOptions (parâmetro do load de um UdfStore)
 * 
 * - selector: string (coluna de agregação)
 * - operation: 'count' | 'sum' | 'max' | 'min' | 'avg' (operador de agregação)
 * 
 */
export class UdfAggregable {
  /** Coluna de agregação da Entidade de trabalho */
  selector: string;
  /** Operação de agregação à ser desempenhada na coluna da Entidade de trabalho  */
  operation: 'count' | 'sum' | 'max' | 'min' | 'avg';
}

File: generators/fullstack-app/templates/2.1.0:7.1.0/<%= projectName %>-backend/src/lib/br/jus/tre_pa/<%= projectName %>-backend/umari/datafilter/core/UdfAggregable.java
package br.jus.tre_pa.sof.umari.datafilter.core;

import com.fasterxml.jackson.annotation.JsonAlias;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.EqualsAndHashCode;

/**
 * Classe com informações para a realização de uma operação de agregação.
 */
@EqualsAndHashCode(of = {"dataField", "operation"})
public class UdfAggregable {

    /**
     * Nome do atributo que sofrerá a operação de agregação.
     */
    @JsonAlias("selector")
    private String dataField;

    /**
     * Tipo de sumarização a ser realizada.
     *
     * @see Operation
     */
    @JsonAlias("summaryType")
    private Operation operation;

    public String getDataField() {
        return this.dataField;
    }

    public Operation getOperation() {
        return this.operation;
    }

    public void setDataField(String dataField) {
        this.dataField = dataField;
    }

    public void setOperation(Operation operation) {
        this.operation = operation;
    }

    public enum Operation {
        @JsonProperty("sum") SUM("SUM(%s) as %s"),
        @JsonProperty("max") MAX("MAX(%s) as %s"),
        @JsonProperty("min") MIN("MIN(%s) as %s"),
        @JsonProperty("count") COUNT("COUNT(%s) as %s"),
        @JsonProperty("dcount") DCOUNT("COUNT(DISTINCT %s) as %s"),
        @JsonProperty("avg") AVG("AVG(%s) as %s");
        String sqlFragment;

        Operation(String sqlFragment) {
            this.sqlFragment = sqlFragment;
        }

        public String getSqlFragment() {
            return sqlFragment;
        }

    }
}
